content/blog/2024-01-11-xilem-backend-roadmap.md

```diff
@@ -0,0 +1,193 @@
+authors = ["Olivier Faure"]
+date = "2024-01-11 15:00:00"
++++
+
+As you may have heard by now, Google Fonts is funding me this year to work on Xilem.
+
+I'm not alone in that: Aaron Muir Hamilton, Daniel McNab and Matt Campbell were funded as well to work on various parts of the ecosystem. I believe this is Matt's third year getting funding from Google Fonts.
+
+Now, *what* I was hired to do might be unclear to you. The scope we agreed on was fairly broad and was stated as "contributing to Xilem's view tree and developer experience", but that could mean a lot of things.
+
+The first thing I want to do is communicate a clearer picture. I have a a lot of plans for Xilem, and they involve major changes from the current architectures, changes that might be worrying to the community at large or even to Raph Levien. I certainly don't want to give the impression that I'm wresting the project away from the Linebender community for Google's interest, and that means being transparent early and often about the things I want to rework.
+
+
+## What I want for Xilem
+
+Xilem is a research-and-development project, the latest in a series (Xi-Core, Druid, Crochet, Lasagna, Idiopath), that aims to explore how UI development can be made compatible with Rust.
+
+This might feel a little banal now, because Rust GUI frameworks are popping up everywhere and starting to settle on similar patterns, but those patterns weren't obvious in 2019! Xilem is very different from Druid, and getting to its current architecture has been a process.
+
+Xilem has been perma-experimental since May 2022. Once it got started, the main Druid maintainers stopped contributing almost entirely with the expectation that Xilem would replace Druid once it was ready. Unfortunately, it still hasn't made enough progress to be on parity with Druid, yet Druid still lies abandoned, which isn't great when people had started relying on it.
+
+It's debatable how much this could have been avoided. As I've pointed out before, the Rust GUI ecosystem is subject to massive [yak-shaving](https://seths.blog/2005/03/dont_shave_that/): many of us came here because we wanted to build a text editor, and now we're all learning about [text rendering](https://faultlore.com/blah/text-hates-you/), [text editing](https://lord.io/text-editing-hates-you-too/), [compositing](https://raphlinus.github.io/ui/graphics/2020/09/13/compositor-ise-vil.html)[^1], [accessibility trees](https://accesskit.dev/accesskit-integration-makes-bevy-the-first-general-purpose-game-engine-with-built-in-accessibility-support/), [using monoids to implement stuff on the GPU](https://raphlinus.github.io/gpu/2021/05/13/stack-monoid-revisited.html), [ECS](https://www.leafwing-studios.com/blog/ecs-gui-framework/), and some concepts that I'm absolutely certain Raph made up like [Bézier paths](https://raphlinus.github.io/curves/2023/04/18/bezpath-simplify.html) and [C++](https://raphlinus.github.io/rust/2023/04/01/rust-to-cpp.html).
```

Comment:
Just noticed that "compositing" link is broken here: `s/compositor-ise-vil/compositor-is-evil`

Reply:
Huh. I noticed it, but I thought the typo came from the original URL. No idea how this typo got here.